app/jobs/push_starter_code_job.rb

```diff
@@ -0,0 +1,10 @@
+class PushStarterCodeJob < ActiveJob::Base
+  queue_as :starter_code
+
+  def perform(creator, assignment_repo_id, starter_code_repo_id)
```

Comment:
I usually try to call a single method on my model from `perform` where possible. That way you've got all of your logic in one place, you can call it synchronously if you need to, the job is only responsible for doing things in the background, and your tests are with the rest of your model code.